src/extern/turbomole.f90

```diff
@@ -163,58 +174,126 @@ subroutine singlepoint(self, env, mol, chk, printlevel, restart, &
    endif
 end subroutine singlepoint
 
-subroutine external_turbomole(env,n,at,xyz,nel,nopen,extcode,extmode,grd,eel,g,dip,lsolv)
+!---------------------------------------------------------
+! To run TURBOMOLE, and cefine if needed 
+!---------------------------------------------------------
+subroutine external_turbomole(env,n,at,xyz,nel,nopen,extcode,extmode,grd,eel,g,dip,lsolv,chrg,uhf)
+   
    use xtb_mctc_accuracy, only : wp
    use xtb_setparam
+   
    implicit none
-   !> Computational environment
+   character(len=*),parameter :: source = 'external_turbomole'
    type(TEnvironment), intent(inout) :: env
-   integer n, at(n), nel, nopen
-   logical grd,lsolv
+      !! Calculation environment to handle I/O stream and error log
+   integer :: n, at(n), nel, nopen
+      !! structural information from mol(Tmolecule) and chk(TRestart) 
+   logical ::  grd
+      !! if gradient calcultion is needed
+   logical :: lsolv
+      !! if solvated
    integer, intent(in) :: extcode, extmode
-   real(wp) xyz(3,n)
-   real(wp) xyz_cached(3,n)
-   real(wp) g  (3,n)
-   real(wp) eel
-   real(wp) dip(3)
+      !! turbomole settings
+   real(wp) :: xyz(3,n)
+      !! coordinates
+   real(wp) :: xyz_cached(3,n)
+      !! to save xyz values
+   real(wp) :: g (3,n)
+      !! gradients
+   real(wp) :: eel
+      !! electronic energy
+   real(wp) :: dip(3)
+      !! dipole moments
+   real(wp), intent(in) :: chrg
+      !! charge
+   integer, intent(in) :: uhf
+      !! multiplicity, number of unpaired electrons
+
    character(len=255) atmp
    character(len=:), allocatable :: syspath, cefine
-   logical :: cache, exist
+   logical :: cache
+      !! to check if molecule moves b/n optimization runs
+   logical :: exist
+   logical :: runCefine
+   integer :: err
 
    cache = .false.
    dip=0
-
+   
+   !> TM input
    inquire(file="control", exist=exist)
+      !! if control file present in the calc dir
    if (.not.exist) then
-      call rdvar("PATH", syspath)
-      call rdpath(syspath, "cefine", cefine, exist)
-      if (exist) then
-         call wrtm(n,at,xyz)
-         call execute_command_line("exec "//cefine//" --func tpss --def2/SVP --cosmo 2.38 --d4 -sym c1 --noopt")
-      end if
+      
+      !> cefine 
+      if(len(set%ext_turbo%input_string).ne.0) then      
+         call rdvar("PATH", syspath)
+         call rdpath(syspath, "cefine", cefine, exist)
+         if (exist) then
+            call wrtm(n,at,xyz)
+            if (allocated(set%ext_turbo%input_string)) then 
+               call execute_command_line("exec "//cefine//" "//set%ext_turbo%input_string)
+            else
+               call execute_command_line("exec "//cefine//" --func tpss --bas def2-SVP --d4 -sym c1 --noopt")
+            endif
+         end if
```

Comment:
Can we maybe omit the usage of cefine by writing our own control files? Something like an internal [TMPrep](https://github.com/fabothch/TMprep)? 

At least, we should probably check if ``cefine`` exists and raise an error if it doesn't (prompting the user to get it).